File: year2/mpp/java/MotorcycleContestHibernate/MotorcycleServer/src/main/java/StartServer.java
import domain.Test;
import domain.User;
import myjdbc.MyJdbcTemplate;
import networking.MotorcycleClientRpcWorker;
import networking.RpcConcurrentServer;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import org.hibernate.Session;
import org.hibernate.Transaction;
import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;
import repositories.database.ParticipantJdbcRepository;
import repositories.database.RaceJdbcRepository;
import repositories.database.TeamJdbcRepository;
import repositories.database.UserJdbcRepository;
import repositories.hibernate.MotorcycleSessionFactory;
import services.MotorcycleService;
import services.MotorcycleServiceImpl;
import validators.ParticipantValidator;
import validators.RaceValidator;
import validators.TeamValidator;
import validators.UserValidator;

public class StartServer {
    private static final Logger LOGGER = LogManager.getLogger(MotorcycleClientRpcWorker.class.getName());

    public static void main(String[] args) {
        RpcConcurrentServer server = null;
        try {
            LOGGER.info("creating server");
            server = getServerXmlConfig();
            server.start();
        } catch (Exception e) {
            LOGGER.warn("creating application context failed " + e);
        } finally {
            if (server != null) {
                server.stop();
            }
        }
    }

    static RpcConcurrentServer getServer() {
        LOGGER.info("creating application context");
        MyJdbcTemplate template = new MyJdbcTemplate("/Users/analazar/Intellij/motorcycle-client-server/MotorcycleServer/src/main/resources/bd.config");
        MotorcycleService service = new MotorcycleServiceImpl(
                new UserJdbcRepository(template, new UserValidator()),
                new RaceJdbcRepository(template, new RaceValidator()),
                new TeamJdbcRepository(template, new TeamValidator()),
                new ParticipantJdbcRepository(template, new ParticipantValidator())
        );
        return new RpcConcurrentServer(55555, service);
    }

    static RpcConcurrentServer getServerXmlConfig() {
        LOGGER.info("creating application context");
        ApplicationContext context = new ClassPathXmlApplicationContext("beans.xml");
        return context.getBean(RpcConcurrentServer.class);
    }
}

File: year2/mpp/c#/MotorcycleClientServer/MotorcycleServer/StartServer.cs
using System;
using MotorcycleCommon.Services;
using MotorcycleContest.Domain.Validators;
using MotorcycleContest.Repositories.Database;
using MotorcycleContest.Repositories.Interfaces;
using MotorcycleContest.Services;
using MotorcycleServer.Networking;

namespace MotorcycleServer
{
    internal class StartServer
    {
        // protected static readonly ILog Logger = LogManager.GetLogger("StartServer");
        
        public static void Main(string[] args)
        {
            RpcConcurrentServer server = null;
            try
            {
                // Logger.Info("creating server");
                server = GetServer();
                Console.WriteLine("server started");
                server.Start();
            }
            catch (Exception exception)
            {
                // Logger.Warn("creating application context failed " + exception.Message);
            }
            finally
            {
                if (server != null)
                {
                    server.Stop();
                }
            }
        }

        private static RpcConcurrentServer GetServer()
        {
            // Logger.Info("creating application context ...");
            IUserRepository userRepository = new UserDbRepository(new UserValidator());
            IRaceRepository raceRepository = new RaceDbRepository(new RaceValidator());
            ITeamRepository teamRepository = new TeamDbRepository(new TeamValidator());
            IParticipantRepository participantRepository = new ParticipantDbRepository(new ParticipantValidator());
            IMotorcycleService service = new MotorcycleServiceImpl(userRepository, raceRepository, teamRepository, participantRepository);
            return new RpcConcurrentServer("127.0.0.1", 55555, service);
        }
    }
}
